### PR TITLE
Add emoji reactions to direct messages

### PR DESCRIPTION
**Pull request by jpangborn**
## Summary

- Adds emoji reaction support to direct messages — participants can react, toggle reactions off, and see reaction chips with counts and tooltips showing who reacted ("You" listed first).
- Introduces `DirectMessageReactionToggled` broadcast event so other participants see reaction updates in real-time via Echo, without the reactor receiving their own event.
- Validates reactions against Spatie's allowed list and reuses the existing `comment` authorization policy for access control.
- Includes feature tests covering add, remove, disallowed input, non-participant authorization, broadcast targeting, tooltip ordering, and out-of-band refresh.

## Release Notes

* **New Features**
  * Added ability to react to direct messages with a reaction picker
  * Reactions display as visual chips with hover tooltips showing reactor names
  * Reaction updates appear in real-time for all conversation participants

* **Tests**
  * Added comprehensive test coverage for direct message reaction functionality

### CODE REVIEW

File: tests/Feature/DirectMessageReactionsTest.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    [$me, , $conversation] = directThread();
+    $comment = $conversation->postComment('<p>private</p>', $me);
+    $stranger = User::factory()->create();
+
+    Livewire::actingAs($stranger)
+        ->test('pages::messages.index')
+        ->set('activeConversationId', $conversation->id)
+        ->call('react', $comment->id, '👍');
+
+    expect($comment->fresh()->reactions()->count())->toBe(0);
```
[the remark this answers is not in the document]
Comment (jpangborn) [reply tests/Feature/DirectMessageReactionsTest.php:76]: Not applicable — the `activeConversation` computed property already scopes queries to `$this->user()->directConversations()`, so a stranger setting `activeConversationId` gets `null` back and `react()` silently early-returns before `authorize()` is ever reached. Adding `assertForbidden()` here would cause the test to fail because there is no 403 — the security boundary is the scoped query, not the policy gate. The side-effect assertion (`reactions()->count() === 0`) correctly verifies the invariant regardless of which layer enforces it.
